test/mochaRunCreator.js

```diff
@@ -9,9 +9,28 @@ function reportError(errorReporter) {
 }
 
 export default function mochaRunCreator(errorReporter = 'process') {
-  return () => {
-    // read: false, not to load file contents
-    gulp.src('src/**/__test__/**/*.js', {read: false})
+  return (file) => {
+    let source = 'src/**/__test__/**/*.js'
+
+    if (file) {
+      // Do not run tests when changed something not JS
+      if (!/\.jsx?$/.test(file.path)) {
```

Comment:
Why just `jsx`? We do not even use this extension in Este (we use `.react.js` instead)